include/libtorrent/settings_pack.hpp

```diff
@@ -1747,6 +1759,22 @@ namespace aux {
 
 		enum suggest_mode_t : std::uint8_t { no_piece_suggestions = 0, suggest_read_cache = 1 };
 
+		enum ssl_version_t : std::uint8_t
+		{
+			// SSL version 3
+			ssl3,
+			// SSL version 2 or 3
+			ssl23,
+			// TLS version 1.1
+			tls11,
+			// TLS version 1.2
+			tls12,
+#if BOOST_VERSION >= 106900
+			// TLS version 1.3
+			tls13,
+#endif
+		};
```

Comment:
I don't think supporting SSL v3 or v2 is a good idea at all. And TLS <= 1.1 is also on its way out (https://hacks.mozilla.org/2020/02/its-the-boot-for-tls-1-0-and-tls-1-1/), but I think completely dropping support for it can be done in a few months from now.

Reply:
my experience (which may be a bit dated by now, but not more than 3 years probably) is that the web server that comes with python 2 *requires* clients to support SSL 2 or 3 (I think that's what the asio value means at least)

Reply:
Well python2 is EOL'd as well but even so, it defaults to disabling SSL 2 and 3, and recommends not using it at all as well:
https://docs.python.org/2/library/ssl.html#ssl-contexts (See the bit about `Changed in version 2.7.16`)
https://docs.python.org/2/library/ssl.html#protocol-versions

Reply:
it seems python supports at least TLS 1.1. Now, how confident are we that no trackers require an old SSL version?

Reply:
@arvidn 
> it seems python supports at least TLS 1.1.

Yes, like I said, totally removing support TLS 1.0 and 1.1 will have to wait for later. Even Firefox is simply doing a "soft" deprecation, where you can still enable usage at the press of a button (see the link above and the release notes for FF 74).

> Now, how confident are we that no trackers require an old SSL version?

Very. If there are any, they should have upgraded long ago anyway.